packages/core/src/__tests__/ceramic.test.ts

```diff
@@ -17,274 +15,118 @@ const seed = u8a.fromString('6e34b2e1a9624113d81ece8a8a22e6e97f0e145c25c1d4d2d0e
-    // ipfs1 <-> ipfs2 <-> ipfs3
-    // ipfs1 <!-> ipfs3
-    await ipfs1.swarm.connect(multaddr2)
-    await ipfs2.swarm.connect(multaddr3)
-
-    const ceramic1 = await createCeramic(ipfs1)
-    const ceramic2 = await createCeramic(ipfs2)
-    const ceramic3 = await createCeramic(ipfs3)
-
-    const controller = ceramic1.context.did.id
-    // ceramic node 2 shouldn't need to have the document open in order to forward the message
-    const doctype1 = await ceramic1.createDocument(DOCTYPE_TILE, { content: { test: 789 }, metadata: { controllers: [controller], tags: ['3id'] } }, { anchor: false, publish: false })
-    const doctype3 = await ceramic3.createDocument(DOCTYPE_TILE, { content: { test: 789 }, metadata: { controllers: [controller], tags: ['3id'] } }, { anchor: false, publish: false })
-    expect(doctype3.content).toEqual(doctype1.content)
-    await ceramic1.close()
-    await ceramic2.close()
-    await ceramic3.close()
-  })
-
-  it('can propagate multiple update across nodes with common connection', async () => {
-    // ipfs1 <-> ipfs2 <-> ipfs3
-    // ipfs1 <!-> ipfs3
-    await ipfs1.swarm.connect(multaddr2)
-    await ipfs2.swarm.connect(multaddr3)
-
-    const ceramic1 = await createCeramic(ipfs1)
-    const ceramic2 = await createCeramic(ipfs2)
-    const ceramic3 = await createCeramic(ipfs3)
+describe('Ceramic', () => {
+    jest.setTimeout(30000)
+    let ipfs: IpfsApi;
+    let ceramic: Ceramic
+    let tmpFolder: any;
 
-    const controller = ceramic1.did.id
+    beforeAll(async () => {
+        tmpFolder = await tmp.dir({ unsafeCleanup: true })
 
-    // ceramic node 2 shouldn't need to have the document open in order to forward the message
-    const doctype1 = await ceramic1.createDocument<TileDoctype>(DOCTYPE_TILE, {
-      content: { test: 321 },
-      metadata: { controllers: [controller], tags: ['3id']},
-      deterministic: true,
+        ipfs = await createIPFS({
+            repo: `${tmpFolder.path}`, config: {
+                Addresses: { Swarm: [`/ip4/127.0.0.1/tcp/${3001}`] }, Bootstrap: []
+            }
+        })
     })
 
-    await anchor(ceramic1)
-    await syncDoc(doctype1)
-
-    // Through a different ceramic instance create a new document with the same contents that will
-    // therefore resolve to the same genesis record and thus the same docId.  Make sure the new
-    // Document object can see the updates made to the first Document object since they represent
-    // the same Document in the network.
-    const doctype3 = await ceramic3.createDocument<TileDoctype>(DOCTYPE_TILE, {
-      content: { test: 321 },
-      metadata: { controllers: [controller], tags: ['3id'] },
-      deterministic: true,
-    }, {
-      anchor: false, publish: false, sync: false
+    afterAll(async () => {
+        await ipfs.stop(() => console.log('IPFS stopped'))
+        await tmpFolder.cleanup()
+        await new Promise(resolve => setTimeout(() => resolve(), 1000))
```

Comment:
I think it's a bit problematic if tests need these kinds of tricks to run, this may hide underlying problems that will show up in consuming apps and libs.

Reply:
these are the same tests that we tried to solve with `ipfs.getConfig()`. IPFS instances are not being stopped correctly. I'll take a look at it in more depth.